packages/sdk/src/browser-wallets/xverse/index.ts

```diff
@@ -40,10 +118,72 @@ async function getAddresses(
  * @returns An object containing `base64` and `hex` if the transaction is not extracted, or `hex` if the transaction is extracted.
  */
 async function signPsbt(
-  _psbt: Psbt,
-  _options: BrowserWalletSignPSBTOptions = {},
+  psbt: Psbt,
+  {
+    finalize = true,
+    extractTx = true,
+    network,
+    inputsToSign,
+  }: XverseSignPSBTOptions = { network: "mainnet", inputsToSign: [] },
 ): Promise<BrowserWalletSignResponse> {
-  throw new OrditSDKError("Method not implemented");
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("Xverse not installed");
+  }
+  if (!psbt || !network || !inputsToSign.length) {
+    throw new OrditSDKError("Invalid options provided");
+  }
+
+  let hex: string;
+  let base64: string | null = null;
+
+  const handleOnFinish = (response: SignTransactionResponse) => {
+    const { psbtBase64 } = response;
+    if (!psbtBase64) {
+      throw new BrowserWalletUserCancelledError("Request canceled by user.");
+    }
+
+    const signedPsbt = Psbt.fromBase64(psbtBase64);
+    if (finalize) {
+      if (!inputsToSign.length) {
+        signedPsbt.finalizeAllInputs();
```

Comment:
actually `!inputsToSign.length` already checked above and will throw error so this line is kindda redundant 🤔

Reply:
Hmm I think this was probably to handle whether inputs are length==0. Let me relook the logic.

Reply:
Removed this `if` block.